doc/reference/kernel/synchronization/semaphores.rst

```diff
@@ -130,3 +130,13 @@ API Reference
 
 .. doxygengroup:: semaphore_apis
    :project: Zephyr
+
+User Mode Semaphore API Reference
+*********************************
+
+The sys_sem exists in user memory working as counter semaphore for user mode
```

Comment:
This isn't much of an explanation.

Unless this is just to tick some FuSa-related box, it would be better to explain that sys_sem is a semaphore IPC object that can live in user memory that uses atomic operations and futex system calls. 

There needs to be a compare/contrast with k_sem, which is also available to user mode with different usage restrictions.

Reply:
> This isn't much of an explanation.

Just copied it verbatim from sys/sem.h as I was not sure what to put here. Please provide better text if you have any. The reason I sent this PR is that I was trying to use sys_sem but noticed that the documentation was missing it even thou the API was documented in .h file.